### ISSUE TITLE
HasOneThrough relationship is... wrong?

### ISSUE THREAD

**Issue by JamShady** [laravel/framework#42114]
- Laravel Version: 9.0.0
- PHP Version: n/a
- Database Driver & Version: n/a

## Description:

_The "has-one-through" relationship defines a one-to-one relationship with another model_, as per the docs. However, the actual example given (according to the DB schema at least) is, _wrong_, and backwards.

The DB schema defines a Mechanic, ala the following - fine.
```
mechanics
    id - integer
    name - string
```

And then it defines a Car, which is associated to a single mechanic via mechanic_id ala:
```
cars
    id - integer
    model - string
    mechanic_id - integer
```
This would represent the 'many' side of a 1:many relationship, i.e. many cars can be associated to a single mechanic, which should yield the following possible relationships:
- $mechanic->cars (via HasMany)
- $car->mechanic (via BelongsTo)

And it's the same with Owner. The Owner is associated to a single car via car_id:
```
owners
    id - integer
    name - string
    car_id - integer
```
Which actually means multiple owners could be related to a single Car, giving rise to the following relationships:
- $car->owners (via HasMany)
- $owner->car (via BelongsTo)

### Problem 1: The schema is backwards
The schema is clearly counter to the intent.

If it was the intention for `cars.mechanic_id` to be a unique field (not mentioned or referenced, and counter-intuitive), then it'd make more sense to define `mechanic.car_id` to associate the Mechanic to a single Car, so via the schema it's obvious which way the 1:* relationship goes.

And same with the Owner. If Car is meant to have only one owner, then `cars.owner_id` should be defined instead of `owners.car_id`.

i.e. the schema ought to be the following if the nature of the relationships the example is attempting to demonstrate is to be defined properly:
```
mechanics
    id - integer
    name - string
    car_id - integer
 
cars
    id - integer
    model - string
    owner_id - integer
 
owners
    id - integer
    name - string
```

### Problem 2: The 'has-one-through' relationship is _wrong_
The documentation presumes that there's a "has-one-through" relationship from the mechanic to the owner, i.e. a mechanic is related through to a single owner. And that's simply not the case according to the schema (unless you accept the incorrect implementation of the foreign fields with uniqueness indexes).

According to the schema, a Mechanic is associated to multiple Cars, and each Car is associated to multiple Owners.

Which means, *each Owner can have a 'has-one-through' relationship to the Mechanic (Owner has one Car, which has one Mechanic), but not the other way around.*

### What should it be?
According to the incorrect schema, Owners have a 'has-one-through' relationship through to a Mechanic, not vice-versa. Owners are associated to a particular Car, which is then associated to a particular Mechanic. Therefore, if we know the Owner, we can determine the Mechanic, but **not the other way around**.

### So what's the problem?
It literally cannot be implemented with the 'HasOneThrough' relationship.

If we try to implement the following:
```
class Owner extends Model
{
    public function mechanic() {
        return $this->hasOneThrough(Mechanic::class, Car::class);
    }
}
```
It won't work because it expects the backwards DB design.

We need to manipulate the key field names to tweak the resulting SQL to get it to work properly, like this:
```
class Owner extends Model
{
    public function mechanic() {
        $mechanic = new Mechanic;
        $car = new Car;
    
        return $this->hasOneThrough(
            $mechanic,
            $car,
            $car->primaryKey,
            $mechanic->primaryKey,
            $car->getForeignKey(),
            $mechanic->getForeignKey()
        );
    }
}
```
This appears to return the correct result, but shouldn't require this much manipulation to achieve

## TLDR
The documentation presumes mechanics are associated to one owner, whereas the schema defines it the other way around. The presumption is incorrect, and the code is engineered around this incorrect assumption.

HasOneThrough is broken, because it primarily works for a broken/incorrect schema implementation, rather than a correctly normalised schema, and therefore does not do (by default) what it's supposed to.

**Comment by driesvints** [comment]
Heya, appreciate the lengthy post but if there's an issue with the documentation please just send in a PR to correct them.
